train_decoder.py

```diff
@@ -289,9 +289,9 @@ def train(
     sample = 0
     samples_seen = 0
     val_sample = 0
-    step = lambda: int(trainer.step.item())
+    step = lambda: int(trainer.steps[0].item())
```

Comment:
offered a helper method https://github.com/lucidrains/DALLE2-pytorch/commit/d7bc5fbedd85564f7050534d06a3d94da50e6c91#diff-617450527162fa367141dbf45e8b201673573479820af0ffc56ba93b7f70947fR537 this would only retrieve the number of training steps for the first unet

Reply:
Will change to using that. I will also have to change the upsampling trainer pr to take advantage of this.

unet 0 is selected arbitrarily for no other reason than it does not matter which unet is chosen and 0 is always present. I am actually just going to remove this variable entirely and replace calls that use it with the helper function since that will be more reliably correct. It will still choose unet 0 though. I could also allow this to be configured, but since it only effects logging that does not seem neccesary.

Reply:
oh ok, as long as you're aware!

Reply:
I do wish there was a more satisfying way to do this, but it really is just an arbitrary choice and as long as the index is consistent between runs it is fine. The way it is handled in the upsampler training is that it takes the first training unet for the log step.

I guess we could go back to the original system of tracking the step separately from the unet step, but unet step is more informative so that doesn't seem like a good idea.